xbmc/windows/GUIWindowScreensaver.cpp

```diff
@@ -53,6 +57,8 @@ EVENT_RESULT CGUIWindowScreensaver::OnMouseEvent(const CPoint &point, const CMou
 
 bool CGUIWindowScreensaver::OnMessage(CGUIMessage& message)
 {
+  auto& context = CServiceBroker::GetWinSystem()->GetGfxContext();
```

Comment:
Not sure I would unconditionally make this call here as you actually need the value only in two out of many `case`s... I would make the call only when actually needed, from within the `case`.

Reply:
For a regular variable I'd totally agree. However for a global I like to "break" least privilege as little as possible and as visibly as possible. See how I put the scope breakage at the beginning of the functions in [ProfileManager.cpp](https://github.com/xbmc/xbmc/blob/master/xbmc/profiles/ProfileManager.cpp#L377). Scope breakage happens all in one stanza there before everything else.